src/NLog/LogFactory.cs

```diff
@@ -254,8 +254,19 @@ public LoggingConfiguration Configuration
                     if (oldConfig != null)
                     {
                         InternalLogger.Info("Closing old configuration.");
-                        Flush();
-                        oldConfig.Close();
+                        if (!FlushAllTargetsSync(oldConfig, DefaultFlushTimeout, ThrowExceptions))
```

Comment:
Funny that the setter of Configuration has changed. it expected to see some changes in Flush(async) or Shutdown methods. 

Could this PR have more side effects?

Reply:
Well it has the side-effect that if flush fails with an exception, then it uses a threadpool thread for closing the configuration. And will continue after 15 secs (Even if a target requires 2 minutes to flush and close).